sam/record.go

```diff
@@ -105,9 +105,12 @@ func IsValidRecord(r *Record) bool {
 // Tag returns an Aux tag whose tag ID matches the first two bytes of tag and true.
 // If no tag matches, nil and false are returned.
 func (r *Record) Tag(tag []byte) (v Aux, ok bool) {
-	for i := range r.AuxFields {
-		if bytes.Compare(r.AuxFields[i][:2], tag) == 0 {
-			return r.AuxFields[i], true
+	if len(tag) != 2 {
```

Comment:
>previous behavior would return false as well.

Only in the case for a non-match where `len(tag) >= 2`. The slice operation in `bytes.Compare` would panic for shorted tag slices.